### PR TITLE
read config from cache on initialize

### PR DESCRIPTION
**Pull request by chantalchung**
- read config from cache if `userId` matches and `fetchDate` is in range

### CODE REVIEW

File: DevCycle/DVCClient.swift [hunk 1]
```diff
@@ -88,41 +88,54 @@ public class DVCClient {
...
+        if let configCacheTTL = options?.configCacheTTL, let disableConfigCache = options?.disableConfigCache, !disableConfigCache {
+            cachedConfig = cacheService.getConfig(user: user, ttlMs: configCacheTTL)
+        }
+        
+        if cachedConfig != nil {
+            self.config?.userConfig = cachedConfig
+            self.isConfigCached = true
+            Log.debug("Loaded config from cache")
+        } else {
+            self.service?.getConfig(user: user, enableEdgeDB: self.enableEdgeDB, extraParams: nil, completion: { [weak self] config, error in
```
Comment (laurawarr) [inline DevCycle/DVCClient.swift:102]: we should still continue to fetch the real config, even if we use the config from the cache

File: DevCycle/DVCClient.swift [hunk 2]
```diff
@@ -88,41 +88,54 @@ public class DVCClient {
...
+        
+        var cachedConfig: UserConfig?
+        if let configCacheTTL = options?.configCacheTTL, let disableConfigCache = options?.disableConfigCache, !disableConfigCache {
+            cachedConfig = cacheService.getConfig(user: user, ttlMs: configCacheTTL)
+        }
+        
+        if cachedConfig != nil {
+            self.config?.userConfig = cachedConfig
+            self.isConfigCached = true
+            Log.debug("Loaded config from cache")
```
Comment (laurawarr) [inline DevCycle/DVCClient.swift:100]: if we're using the cached config, we should trigger the callbacks associated with a config change (ie. variable.onUpdate)

File: DevCycle/DVCClient.swift [hunk 3]
```diff
@@ -88,41 +88,54 @@ public class DVCClient {
...
+                guard let self = self else { return }
+                if let error = error {
+                    Log.error("Error getting config: \(error)", tags: ["setup"])
+                    self.cache = self.cacheService.load()
+                } else {
+                    if let config = config {
+                        Log.debug("Config: \(config)", tags: ["setup"])
+                    }
+                    self.config?.userConfig = config
+                    self.isConfigCached = false
```
Comment (laurawarr) [inline DevCycle/DVCClient.swift:112]: could we move this to be set to false any time a config is fetched? as opposed to just in the init

Comment (chantalchung) [reply DevCycle/DVCClient.swift:112]: Not sure how to set this value within the `RequestConsolidator` (where config is fetched) but I can also set it for `identify`, `reset`?

File: DevCycle/DVCClient.swift (hunk 2 above)
Comment (chantalchung) [reply DevCycle/DVCClient.swift:100]: hmm what callback do we have to trigger?

Comment (laurawarr) [reply DevCycle/DVCClient.swift:100]: I'm not sure exactly, I'll poke through the getConfig code and see if I can find it as well. When `variable.onUpdate` is called a a handler is registered, and we need to find what calls those handlers when the config changes

Comment (chantalchung) [reply DevCycle/DVCClient.swift:100]: In the `DVCVariable` class it seems `addNotificationObserver()` is called on initialization of the variable and then in `propertyChange()` it handles updating the variables. I don't see anything in getConfig though

**Review by laurawarr — APPROVED** [review]

**Pull request merged**